Improve test for the next_execution cli command

mdediana opened the pull request:
Improve `TestCliDags.test_next_execution` in terms of efficiency and readability. ~~Perhaps the `@pytest.mark.quarantined` decorator could be removed, but it is still there because locally the test takes longer than others in the same test case.~~  The `@pytest.mark.quarantined` decorator has been removed.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

mdediana committed:
Improve test for the next_execution cli command

It still tests the same functionality, but it is now more efficient and
readable. The changes are:

 - The DB is cleaned only once instead of eight times
 - Use a fixed datetime instead of timezone.utcnow (deterministic)
 - Use redirect_stdout instead of subprocess
 - Clean up pylint warning
 - Test None output once instead of four times

turbaszek commented on tests/cli/commands/test_dag_command.py:
```diff
@@ -240,21 +239,30 @@ def test_cli_backfill_depends_on_past_backwards(self, mock_run):
+        dr = session.query(DagRun).filter(DagRun.dag_id.in_(dag_ids))
+        dr.delete(synchronize_session=False)
+        session.commit()
+        session.close()
```
Let's use `create_session` context manager, WDYT?

turbaszek reviewed (approved):
LGTM! Additonally can you:
- use `DagRunType.MANUAL.value` instead of `run_id='manual_...` at L291?
- remove the DagRun that is created during test?

mdediana committed:
Address PR comments

Issues addressed:

 - Use create_session

 - Use DagRunType.MANUAL.value

 - Remove DagRuns created by the test

turbaszek commented on tests/cli/commands/test_dag_command.py:
```diff
@@ -240,21 +240,29 @@ def test_cli_backfill_depends_on_past_backwards(self, mock_run):
 
     @pytest.mark.quarantined
```
```suggestion

```
Should we try?

mdediana replied:
Sure!

mdediana committed:
Remove test from quarantine

Co-authored-by: Tomek Urbaszek <turbaszek@gmail.com>
